src/Arriba/Arriba.Server/Middlewares/ArribaNormalizeIdentityMiddleware.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace Arriba.Server.Middlewares
+{
+    public class ArribaNormalizeIdentityMiddleware
+    {
+        private const string ClaimTypeName = "name";
+        private readonly RequestDelegate _next;
+
+        public ArribaNormalizeIdentityMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            if (httpContext.User.Identity.IsAuthenticated)
+            {
+                var user = NormalizeIdentity(httpContext.User);
+                httpContext.User = user;
+            }
+
+            await _next(httpContext);
+        }
+
+        private ClaimsPrincipal NormalizeIdentity(ClaimsPrincipal user)
+        {
+            var claims = new List<Claim>();
+            string name = string.Empty;
+
+            foreach (var claim in user.Claims)
+            {
+                bool addClaim = true;
+
+                if (claim.Type == ClaimTypeName)
+                {
+                    claims.Add(new Claim(ClaimTypes.Name, claim.Value));
+                    name = claim.Value;
+                }
+
+                if (claim.Type == ClaimTypes.Role)
+                {
+                    addClaim = false;
+                    claims.Add(new Claim(ClaimTypes.Role, claim.Value.ToLowerInvariant()));
```

Comment:
Why use ToLowerInvariant()

Reply:
Because it will depend in how the crawler or the hack deals with the database security groups, it should me be Lower() as the Hack is doing.

Reply:
Shouldn't let that affect this. If the values are case sensitve then use a constant to replace the value. If case sensitivity does not matter then if this breaks, fix the bug elsewhere in the code to not be case dependent.

Reply:
I changed the class SecurityIdentity, it was changing the Identity name to ToLowerInvariant() and now I'm using the name as received, also changed the tests that was impacted.